docs/architecture/adr.rst

```diff
@@ -439,6 +439,190 @@ Conclusion
 For all backends performance result are pretty similar to ``Accumulo``
 and ``Cassandra`` backend numbers. In order not to duplicate data these
 numbers were omitted. Thread pool size mostly depend on CPU cores
-availble, less on RAM. In order not to loose performane should not be
-used threads more than CPU cores availble for java machine, otherwise
+available, less on RAM. In order not to loose performance should not be
+used threads more than CPU cores available for java machine, otherwise
 that can lead to significant performance loss.
+
+0004 - ETL Pipelne
+------------------
+
+Context
+^^^^^^^
+
+Current GeoTrellis ETL does not allow to determine ETL as a pipeline of transformation / actions.
+This document would describe a new approach(inspired by [PDAL Pipeline](https://www.pdal.io/pipeline.html))
+with a new ETL JSON description.
+
+Decision
+^^^^^^^^
+
+We can divide current ETL into following steps:
```

Comment:
`divide current` -> `divide the current`

`into following` -> `into the following`